src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,201 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.enums.CommandWord;
+import duke.enums.Message;
+import duke.exception.DukeException;
+
+import java.util.StringTokenizer;
+
+/**
+ * Parses the user input to determine what command the user intends to make
+ */
+public class Parser {
+    
+    private static final String DEADLINE_DELIMITER = "/by";
+    private static final String EVENT_DELIMITER = "/at";
+    private static final String TIME_DELIMITER = "-";
+    
+    
+    /**
+     * Returns the correct command after parsing the user's input
+     *
+     * @param input user's input
+     *
```

Comment:
Might not be necessary to add a blank line to separate the parameters, return values and exceptions.

Reply:
Yeap thanks for pointing this out. I think this is a non-issue as it's not really mentioned in the code conventions: 
![image](https://user-images.githubusercontent.com/38996397/91745792-209b1680-ebee-11ea-8e2b-6593ee04f379.png) 

It keeps things neater when there are multiple params, exceptions.. to group and it's an automated formatting using IntelliJ's codestyle, so I think I'll not be changing this. Thanks though :+1: